app/grandchallenge/evaluation/views/api.py

```diff
@@ -23,3 +46,118 @@ class EvaluationViewSet(ReadOnlyModelViewSet):
         *api_settings.DEFAULT_RENDERER_CLASSES,
         PaginatedCSVRenderer,
     )
+
+    @action(
+        detail=True, methods=["PATCH"], permission_classes=[CanClaimEvaluation]
+    )
+    def claim(self, request, *args, **kwargs):
+        evaluation = self.get_object()
+
+        if not evaluation.status == Evaluation.PENDING:
+            return Response(
+                {"status": "You can only claim pending evaluations."},
+                status=400,
+            )
+
+        if request.user.claimed_evaluations.exists():
+            return Response(
+                {"status": "You can only claim one evaluation at a time."},
+                status=400,
+            )
+
+        evaluation.status = Evaluation.CLAIMED
+        evaluation.claimed_by = request.user
+        evaluation.started_at = now()
+        evaluation.save()
+
+        serializer = ExternalEvaluationSerializer(
```

Comment:
The serializer is not checked to see if it is valid, and is also not saved. I would expect to see something like:

```python
        object = self.get_object()

        ...

        serializer = self.get_serializer(
            instance=object, data=request.data, partial=True
        )

        if serializer.is_valid():
            serializer.save()
            return Response(data=serializer.data)
        else:
            return Response(
                data=serializer.errors, status=HTTP_400_BAD_REQUEST
            )
```

Reply:
Ok, what is the best place to set `claimed_by` and `started_at` and `status` then? In the `update` method on the serializer?

Reply:
Yes - serializers are the DRF equivalent of forms, so any modification of the objects should be done there.